### PR TITLE
Stub vscode 'Tests' API

### PR DESCRIPTION
**Pull request by alvsan09**
#### What it does

This enables the possibility to load and start plugins which depend on the 'Tests' API.
  The 'Tests' functionality will not be operational but the rest of
the functionality offered by the plugin will be.

#### How to test

1. Update the supported `vscode` API version in `theia` to `1.68.0`, at: 
 > dev-packages/application-package/src/api.ts 
2. Reproduce the problem in the `theia` instance before switching to this PR, by installing the following `vscode` sample: https://github.com/alvsan09/vscode-extension-samples/releases/download/test-provider-prel-0.0.1/test-provider-sample-0.0.1.vsix
 - Open a workspace folder containing `markdown` files (extension `.md`)
 - you shall see the following failure message:
image

2. Update your `theia` repository to use this PR
  - Open the same workspace and make sure the message above is no longer displayed and the '.md' files open fine.

A more realistic test can be achieved using the latest `ms-python.python` extension, version `2022.14.0.`,
however additional API support is needed in addition to this change.
A test branch is provided for this purpose: 

1. Update your `theia` repository to load this PR on top of the branch offering additional missing API as follows:
> git fetch
> git checkout asl/tests_api_python_test
> git cherry-pick asl/test_api_stub
build it and start it.

2. Download the latest python extension via `openVSX` registry: 
https://open-vsx.org/api/ms-python/python/2022.14.0/file/ms-python.python-2022.14.0.vsix

3. Using the extensions view install the `vsix` above
4. The extension will attempt to install 4 python related extensions, the ones related to `jupyter` notes will not be successful however the core `python` extension will install and be functional, see the following video clip.

https://user-images.githubusercontent.com/76971376/193086561-863cf633-3af3-4342-904a-988e071cee6d.mp4

The stubbed API has been marked in`theia.d.ts` with the special tag `@stubbed` in order to reflect it is as such in the report generated by the `eclipse-theia/vscode-theia-comparator`

See a portion of the generated report when running it locally: 
image

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin/src/theia.d.ts
Comment (colin-grant-work) [inline packages/plugin/src/theia.d.ts:12244]: Should we mark this as stubbed? I see you've put the implementation in the `types-impl` file rather than the `stubs` area, and the values are correct, so the enum itself is completely implemented, and a future implementer of the full API wouldn't need to do any more work here. Same question re: `TestTag`, `TestRunRequest`, and `TestMessage`.

**Comment by JonasHelming** [comment]
Relates to https://github.com/eclipse-theia/theia/issues/10669

File: packages/plugin/src/theia.d.ts
Comment (alvsan09) [reply packages/plugin/src/theia.d.ts]: > Should we mark this as stubbed? I see you've put the implementation in the `types-impl` file rather than the `stubs` area, and the values are correct, so the enum itself is completely implemented, and a future implementer of the full API wouldn't need to do any more work here. Same question re: `TestTag`, `TestRunRequest`, and `TestMessage`.

@colin-grant-work, I tried the way you propose, it seems clear enough in the report since the main triggers remain marked as `@stubbed`, 
So I posted a new commit reflecting your proposal.

Thanks!!

**Comment by JonasHelming** [comment]
@alvsan09 Is this ready for a re review?

**Comment by alvsan09** [comment]
> @alvsan09 Is this ready for a re review?
Yes, I had missed requesting it, thanks for catching it !!

**Review by colin-grant-work — APPROVED** [review]
The implementations appear sound for stubs (where applicable) and the plugins activate successfully, and the API is correctly evaluated as either newly supported or newly stubbed by the comparator.

**Comment by JonasHelming** [comment]
@alvsan09 Let's merge this if possible :-)

**Comment by alvsan09** [comment]
The commits were squashed and then re-based to latest master just before merging.

**Comment by alvsan09** [comment]
> The implementations appear sound for stubs (where applicable) and the plugins activate successfully, and the API is correctly evaluated as either newly supported or newly stubbed by the comparator.

@colin-grant-work , thanks for the review and testing !!

**Comment by alvsan09** [comment]
> @alvsan09 Let's merge this if possible :-)
@JonasHelming, Done !!

**Pull request merged**
